### ISSUE TITLE
iOS Build Issue

### ISSUE THREAD

**Issue by caseyg1204** [getbouncer/react-native-cardscan#56]
Hello - I am having trouble getting this project to build

```Undefined symbols for architecture x86_64:
  "(extension in Vision):__C.VNRecognizedText.boundingBox(for: Swift.Range<Swift.String.Index>) throws -> __C.VNRectangleObservation?", referenced from:
      closure #1 (__C.VNRecognizedTextObservation) -> CardScan.OcrObject? in closure #1 (__C.VNRequest, Swift.Error?) -> () in static CardScan.AppleOcr.performOcr(image: __C.CGImageRef, completion: ([CardScan.OcrObject]) -> ()) -> () in libCardScan.a(AppleOcr.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)```

`"react-native-cardscan": "^1.1.11",`

react-native: 0.63.2

xCode 12.0.1

**Comment by jaimejiyepark** [comment]
Hello @caseyg1204!
Thank you for bringing up this issue! 
Can I ask if you were using a simulator or a real device?
If you were using a real device, can you tell me the model and the iOS version on it?

**Comment by caseyg1204** [comment]
Hi @jaimejiyepark , thanks for your help. It was a build time error for simulator.

**Comment by jaimejiyepark** [comment]
Great! I will be closing this issue now. Thank you again bringing to our attention.

**Comment by caseyg1204** [comment]
@jaimejiyepark why did you close it? I am unable to build project if I include your project

**Comment by jaimejiyepark** [comment]
@caseyg1204 I am so sorry. I misinterpreted your response. Let me look into it right now.

**Comment by caseyg1204** [comment]
thanks @jaimejiyepark . apologies for not being clear

**Comment by jaimejiyepark** [comment]
Hey @caseyg1204 , 
I can't seem to replicate the error on my end. 
Can I some detail on the steps you took to build your project while including cardscan?
Also can I see what your Podfile looks like?

**Comment by caseyg1204** [comment]
Hi @jaimejiyepark I created a sample project that is having same issue: https://github.com/caseyg1204/cardScan

**Comment by jaimejiyepark** [comment]
@caseyg1204  Thank you so much! I have been able to reproduce the error with your example. Let me get back to you once I have a solution.

**Comment by jaimejiyepark** [comment]
Hey @caseyg1204 ,
I just wanted to give you an update on what is going on. I have a strong hunch that this error is due to a build config rule I added for Carthage support where it excludes certain architectures for XCode12 simulators. I am still working on the solution but please check out our [troubleshooting page](https://docs.getbouncer.com/card-scan/react-native-integration-guide/troubleshooting). I will give you another update very soon!

**Comment by jaimejiyepark** [comment]
Hello @caseyg1204 ,
I hope you are doing well. I believe I found the solution to this issue! It takes a couple of steps but fortunately they are issues we have tackled before listed in our [troubleshooting page](https://docs.getbouncer.com/card-scan/react-native-integration-guide/troubleshooting).

1. Dealing with error `ld: symbol(s) not found for architecture x86_64`
    * We have multiple solutions for this error but the one I tested and successfully built the app with was to add `use_framework!` in the `Podfile`. Please make sure to check out our [troubleshooting page](https://docs.getbouncer.com/card-scan/react-native-integration-guide/troubleshooting) for the full set of instructions since there are more steps to this solution. Your `Podfile` should look something like this:
```swift
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '10.0'
use_frameworks!

target 'cardScan' do
  config = use_native_modules!
  
  use_react_native!(:path => config["reactNativePath"])

  target 'cardScanTests' do
    inherit! :complete
    # Pods for testing
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable these next few lines.
  # use_flipper!
  # post_install do |installer|
  #  flipper_post_install(installer)
  # end
end

target 'cardScan-tvOS' do
  # Pods for cardScan-tvOS

  target 'cardScan-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end
end
```
2. Dealing with errors such as `ld: warning: Could not find auto-linked library 'swiftCoreMl'`
    * This was resolved by creating a bridge header then re-building. The instructions to make one is listed [here](https://stackoverflow.com/questions/52536380/why-linker-link-static-libraries-with-errors-ios/56176956#56176956).

3. If you are building the app with a simulator through react-native (i.e. `react-native run-ios`) , make sure to go into the `ios/.xcworkspace` and build in XCode successfully before running the app with react-native. 

I hope this helps! If you have any issues please let me know I will get back to you as soon as possible. Please make sure to head to our [troubleshooting page](https://docs.getbouncer.com/card-scan/react-native-integration-guide/troubleshooting) for more detailed instructions.

**Comment by jaimejiyepark** [comment]
Oh! Also small hiccup in our README:
* The way to configure the API key in `AppDelegate.m` is actually `[ScanViewController configureWithApiKey:@"<your_api_key_here>"];`

Sorry about the confusion!

**Comment by awushensky** [comment]
@caseyg1204 I'm going to close this issue since it's dormant. Please feel free to re-open it if your issue is not solved.

**Comment by DKetaki** [comment]
![image](https://user-images.githubusercontent.com/39909659/108099933-3ffa0300-70ab-11eb-82b8-f17d656841f6.png)


Hello @jaimejiyepark  I am also getting the same error. Its been two days and I have tried everything as mentioned in the doc. 
I can't use `use_framework!` as I have some `static framework ` linking.

Therefore opening this issue.

**Comment by jaimejiyepark** [comment]
Hello @DKetaki ,
Let me take a look into this. In the meanwhile, do you mind sharing some details about your build?
* An example of what your Podfile looks like
* Rect-native version, react-native-cardscan version
* XCode version

**Comment by DKetaki** [comment]
Please find the below details.

**Comment by jaimejiyepark** [comment]
Thank you for the details. Let me try to reproduce this build issue and I'll get back to you.

**Comment by DKetaki** [comment]
Hello @jaimejiyepark ,any update on this issue.

**Comment by maranmanicse** [comment]
@jaimejiyepark , Any update this issue. Can i implement this for both ios and android?

**Comment by jaimejiyepark** [comment]
Hello @DKetaki,
I was actually OOO last week and passed on this issue to another team member but it seems as though they haven't gotten to it yet. My apologies. I am back now and I will pick up where I left off. Let me give you an update of my progress EOD.

cc @maranmanicse

**Comment by jaimejiyepark** [comment]
@DKetaki I can't seem to replicate your issue on my end. Please let me know of your thoughts of some steps we can take that I will list below. We can then decide together how to proceed. 

1. We can deploy an updated `react-native-cardscan` version. You can test the new version on your project and see if the issue persists. 
2.  You can provide me an example app with the replicated issue and I can try to debug with your example.

**Comment by jaimejiyepark** [comment]
@DKetaki We have released a new version `react-native-cardscan@1.1.27` which includes the updated `CardScan@2.0.9` library. The main changes to the deploy are as follows:
* The scanning UI is changed to a much simplified UI 
* The library is gated from iOS 11.2 since our expiration extraction ML model requires at least iOS 11.2

<img src="https://user-images.githubusercontent.com/42391849/109235117-e4ea9f00-7781-11eb-9ed8-c610e00de7aa.PNG" height="500" />

Please give this version a try and if the issue persists, we can continue from there. Hope this works out and please keep me updated. 

cc @maranmanicse

**Comment by maranmanicse** [comment]
Thank you
